datumaro/util/__init__.py

```diff
@@ -196,16 +196,12 @@ def wrapped_func(*args, **kwargs):
             if implicit:
                 fglobals = func.__globals__
 
-                has_arg = arg_name in fglobals
-                old_val = fglobals.get(arg_name)
+                old_val = fglobals[arg_name]
```

Comment:
Now it fails with a cryptic error if a variable is not defined. If we accept this solution, I suggest to add docs or raise a more clear error with a possible solution.